transcendence-app/src/user/user.service.ts

```diff
@@ -250,4 +249,80 @@ export class UserService {
     }
     return friends;
   }
+
+  async getUserWithRolesFromUsername(
+    username: string,
+  ): Promise<UserWithRoles | null> {
+    const user = await this.userRepository.getByUsername(username);
+    if (!user) {
+      throw new UnprocessableEntityException();
```

Comment:
suggestion: For the get and delete methods, we could throw a `NotFoundException()` when there is no user with that `id`/`username`

Reply:
done